pkg-new/replicatedapi/client.go

```diff
@@ -144,30 +150,45 @@ func (c *client) newRetryableRequest(ctx context.Context, method string, url str
 
 // injectHeaders injects the basic auth header, user agent header, and reporting info headers into the http.Header.
 func (c *client) injectHeaders(header http.Header) {
-	header.Set("Authorization", "Basic "+basicAuth(c.license.Spec.LicenseID, c.license.Spec.LicenseID))
+	if c.license.IsEmpty() {
```

Comment:
not all of these headers depend on a license

Reply:
But don't we need auth?